src/imitation/rewards/serialize.py

```diff
@@ -0,0 +1,99 @@
+"""Load serialized policies of different types."""
+
+from typing import Callable
+
+import numpy as np
+from stable_baselines.common.vec_env import VecEnv
+
+from imitation.rewards import discrim_net, reward_net
+from imitation.util import registry, util
+
+RewardFn = Callable[[np.ndarray, np.ndarray, np.ndarray], np.ndarray]
+RewardLoaderFn = Callable[[str, VecEnv], RewardFn]
+RewardNetLoaderFn = Callable[[str, VecEnv], reward_net.RewardNet]
+
+reward_net_registry: registry.Registry[RewardNetLoaderFn] = registry.Registry()
+reward_registry: registry.Registry[RewardLoaderFn] = registry.Registry()
+
+
+def _add_reward_net_loaders(classes):
+  for name, cls in classes.items():
+    loader = registry.build_loader_fn_require_path(cls.load)
+    reward_net_registry.register(key=name, value=loader)
+
+
+REWARD_NETS = {
+    "BasicRewardNet": reward_net.BasicRewardNet,
+    "BasicShapedRewardNet": reward_net.BasicShapedRewardNet,
+}
+_add_reward_net_loaders(REWARD_NETS)
+
+
+def _load_discrim_net(cls):
+  def f(path: str, venv: VecEnv):
+    discrim_net = cls.load(path)
+    # TODO(gleave): expose train reward as well? (hard due to action probs?)
+    return discrim_net.reward_test
```

Comment:
Maybe Discrim net builds its own sess

Reply:
I think it actually should have session creation logic too, added. (It worked because it used the default session, but in general this can cause name clashes.)